docs/source/advanced_usage.rst

```diff
@@ -8,128 +8,23 @@ Advanced Usage
-   in exactly one of these files.
-3. The default values specified in ``pyquil.api._config``. These should not be changed by any Pyquil
-   user, as doing so may lead to difficult-to-debug behavior.
-
-These are all the configuration options available to you, and where they can be set:
-
-.. list-table:: PyQuil Configuration Options
-   :widths: 50 50 50
-   :header-rows: 1
-
-   * - Setting
-     - Environment
-     - Configuration File
-   * - Forest Server URL
-
-       Source of device information
-     - ``FOREST_URL``
-     - ``.qcs_config``
-
-       Key: ``url``
-   * - Dispatch URL
-
-       Provides QPU engagement authorization
-     - ``FOREST_DISPATCH_URL``
-     - ``.qcs_config``
-
-       Key: ``dispatch_url``
-   * - User Authentication Token Path
-
-       File path to the authentication token
-       obtained from `qcs <https://qcs.rigetti.com/auth/token>`_.
-     - ``USER_AUTH_TOKEN_PATH``
-     - ``.qcs_config``
-
-       Key: ``user_auth_token_path``
-   * - QCS URL
-
-       QCS website
-     - ``QCS_URL``
-     - ``.qcs_config``
-
-       Key: ``qcs_url``
-   * - QPU URL
-
-       Send binaries to the QPU
-     - ``QPU_URL``
-     - ``.forest_config``
-
-       Key: ``qpu_endpoint_address``
-   * - QVM URL
-
-       Simulator
-     - ``QCS_URL``
-     - ``.forest_config``
-
-       Key: ``qvm_address``
-   * - QPU Compiler URL
-
-       Send native Quil and receive a binary
-     - ``QPU_COMPILER_URL``
-     - ``.forest_config``
-
-       Key: ``qpu_compiler_address``
-   * - Local Compiler URL
-
-       Send Quil and receive native Quil
-     - ``QUILC_URL``
-     - ``.forest_config``
-
-       Key: ``qpu_compiler_address``
-
-The configuration options omitted from this table but present in ``pyquil.api._config`` are
-deprecated and should not be used.
+.. _pyquil_configuration:
+
+pyQuil Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+:py:class:`~pyquil.api.QCSClientConfiguration` instructs pyQuil on how to connect with the
+components needed to compile and run programs (quilc, QVMs, and QCS). Any APIs that take a configuration object
+as input (e.g. :func:`get_qc()`) typically do so optionally, so that a default configuration can be loaded for
+you if one is not provided. You can override this default configuration by either instantiating your own
+:py:class:`~pyquil.api.QCSClientConfiguration` object and providing it as input to the function in question,
+or by setting the ``QCS_SETTINGS_FILE_PATH`` and/or ``QCS_SECRETS_FILE_PATH`` environment variables to have
+pyQuil load its settings and secrets from specific locations. By default, configuration will be loaded from
+``$HOME/.qcs/settings.toml`` and ``$HOME/.qcs/secrets.toml``.
+
+Additionally, you can override whichever QVM and quilc URLs are loaded from ``settings.toml`` by setting the
+``QCS_SETTINGS_APPLICATIONS_PYQUIL_QVM_URL`` and/or ``QCS_SETTINGS_APPLICATIONS_PYQUIL_QUILC_URL`` environment
+variables.
```

Comment:
I would mention the default values assume the QVM and Quilc to be running locally at ports 5555 and 5000 (double check me on that), respectively.